File: week5/Java/225. Implement Stack using Queues.java
class MyStack {

    private Queue<Integer> q;

    /** Initialize your data structure here. */
    public MyStack() {
        q = new ArrayDeque<>();
    }
    
    /** Push element x onto stack. */
    public void push(int x) {
        q.add(x);
        var i=1;
        while(i < q.size()){
            q.add(q.remove());        
            i++;
        }
    }
    
    /** Removes the element on top of the stack and returns that element. */
    public int pop() {
        return q.remove();
    }
    
    /** Get the top element. */
    public int top() {
        return q.peek();
    }
    
    /** Returns whether the stack is empty. */
    public boolean empty() {
        return q.isEmpty();
    }
}

File: week5/python/225. Implement Stack using Queues.py
class MyStack:

    def __init__(self):
        
        self.stack = [] #creating an empty list(stack)

    def push(self, x: int) -> None:
        
        self.x = x
        self.stack.append(self.x) #adding an element using append

    def pop(self) -> int:
        
        return self.stack.pop()  #removing the last Element Using pop

    def top(self) -> int:
        
        return self.stack[-1]

    def empty(self) -> bool:
        
        return len(self.stack) == 0 #if stack is empty


